app/src/main/java/org/simple/clinic/patient/recent/RecentPatientsViewController.kt

```diff
@@ -0,0 +1,49 @@
+package org.simple.clinic.patient.recent
+
+import io.reactivex.Observable
+import io.reactivex.ObservableSource
+import io.reactivex.ObservableTransformer
+import io.reactivex.rxkotlin.ofType
+import org.simple.clinic.ReplayUntilScreenIsDestroyed
+import org.simple.clinic.ReportAnalyticsEvents
+import org.simple.clinic.facility.FacilityRepository
+import org.simple.clinic.patient.PatientRepository
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.widgets.ScreenCreated
+import org.simple.clinic.widgets.UiEvent
+import javax.inject.Inject
+
+typealias Ui = RecentPatientsView
+typealias UiChange = (Ui) -> Unit
+
+class RecentPatientsViewController @Inject constructor(
+    private val userSession: UserSession,
+    private val patientRepository: PatientRepository,
+    private val facilityRepository: FacilityRepository
+) : ObservableTransformer<UiEvent, UiChange> {
+
+  override fun apply(events: Observable<UiEvent>): ObservableSource<UiChange> {
+    val replayedEvents = ReplayUntilScreenIsDestroyed(events)
+        .compose(ReportAnalyticsEvents())
+        .replay()
+
+    return Observable.mergeArray(fetchRecentPatients(replayedEvents))
+  }
+
+  private fun fetchRecentPatients(events: Observable<UiEvent>): Observable<UiChange> =
+      events.ofType<ScreenCreated>()
+          .flatMap { userSession.requireLoggedInUser() }
+          .map(facilityRepository::currentFacilityUuid)
+          .flatMap { facilityUuid -> patientRepository.recentPatients(facilityUuid, 10) }
+          .map { recentPatients ->
+            val recentPatientItems = recentPatients.map {
+              RecentPatientItem(
+                  name = it.fullName,
+                  age = it.age?.value ?: -1,
```

Comment:
Why are we setting `age` as -1 here?
If there is no age present we should check `dateOfBirth`